os_net_config/impl_ifcfg.py

```diff
@@ -309,56 +303,58 @@ def iproute2_apply_commands(self, device_name, filename, data):
+               changes['IPADDR'] == 'modified':
                 if old_cidr:
-                    commands.append("addr del %s/%s dev %s" %
-                                    (file_values["IPADDR"], old_cidr,
-                                     device_name))
+                    ip = file_values['IPADDR']
+                    commands.append(f'addr del {ip}/{old_cidr} dev {dev_name}')
                 else:
                     # Cannot remove old IP specifically if netmask not known
-                    commands.append("addr flush dev %s" % device_name)
-            if changes["IPADDR"] == "added" or changes["IPADDR"] == "modified":
-                commands.insert(0, "addr add %s/%s dev %s" %
-                                (data_values["IPADDR"], new_cidr, device_name))
-        if "MTU" in changes:
-            if changes["MTU"] == "added" or changes["MTU"] == "modified":
-                commands.append("link set dev %s mtu %s" %
-                                (device_name, data_values["MTU"]))
-            elif changes["MTU"] == "removed":
-                commands.append("link set dev %s mtu 1500" % device_name)
-        if "DOMAIN" in changes:
-            # Remove searchdomains from /etc/resolv.conf
-            changes.append("sed -i '/search/d' /etc/resolf.conf")
-            # Add searchdomains to /etc/resolv.conf
-            if changes["DOMAIN"] == "added" or changes["DOMAIN"] == "modified":
-                changes.append("echo 'search %s' >> /etc/resolv.conf" %
-                               data_values["DOMAIN"])
-        if "DNS1" in changes:
+                    commands.append(f'addr flush dev {dev_name}')
+            if changes['IPADDR'] == 'added' or changes['IPADDR'] == 'modified':
+                ip = data_values['IPADDR']
+                commands.insert(0, f'addr add {ip}/{new_cidr} dev {dev_name}')
+        if 'MTU' in changes:
+            if changes['MTU'] == 'added' or changes['MTU'] == 'modified':
+                mtu = data_values['MTU']
+                commands.append(f'link set dev {dev_name} mtu {mtu}')
+            elif changes['MTU'] == 'removed':
+                commands.append(f'link set dev {dev_name} mtu 1500')
+        if 'DOMAIN' in changes:
+            old = file_values['DOMAIN']
+            new = data_values['DOMAIN']
+            if changes['DOMAIN'] == 'added':
+                commands.append(f"echo 'search {new}' >> /etc/resolv.conf")
+            elif changes['DOMAIN'] == 'modified':
+                commands.append(f"sed -i -e 's/{old}/{new}/g' "
+                                "/etc/resolv.conf")
+            elif changes['DOMAIN'] == 'removed':
+                commands.insert("sed -i '/search/d' /etc/resolv.conf")
+        if 'DNS1' in changes:
             # Remove nameservers from /etc/resolv.conf
-            changes.append("sed -i '/nameserver/d' /etc/resolv.conf")
+            commands.append("sed -i '/nameserver/d' /etc/resolv.conf")
```

Comment:
There is an issue with the method you are using, and that's that the "sed -i '/^nameserver[]/d resolv.conf" will remove all nameservers. So if we go from two nameservers to one it could cause issues. Rather than that if either one changes I will remove both and then add one or both from the new data.